compiler/test/dotty/tools/dotc/CompilationTests.scala

```diff
@@ -189,7 +189,6 @@ class CompilationTests extends ParallelTesting {
     compileFile("../tests/neg-custom-args/xfatalWarnings.scala", defaultOptions.and("-Xfatal-warnings")) +
     compileFile("../tests/neg-custom-args/i3561.scala", defaultOptions.and("-Xfatal-warnings")) +
     compileFile("../tests/neg-custom-args/pureStatement.scala", defaultOptions.and("-Xfatal-warnings")) +
-    compileFile("../tests/neg-custom-args/i3589-a.scala", defaultOptions.and("-Xfatal-warnings")) +
```

Comment:
Nice! We support emitting switch on value classes. Maybe open an issue to figure out if this is expected or not.